### PR TITLE
Bump chisel3 and firrtl - Async reset support

### PR DESCRIPTION
**Pull request by jackkoenig**
This is a big one, buckle up folks

#### Features
* Add Stage/Phase APIs
* Add "Aspect Oriented"-inspired API support
* Add native `AsyncReset` and abstract `Reset` support
* Add utility functions `unsignedBitLength` and `signedBitLength`
* Move `dontTouch`, `RawModule`, and `MultiIOModule` out of `chisel3.experiemental` to `chisel3`
* Add `reduceTree` to `Vec`
* Strip outer objects and classes from naming of Modules

#### Bugfixes
* Support `withClock` in `RawModules` with no implicit reset
* Support bulk connecting `DontCare` to Bundles with `Analog` fields
* Make LFSR support async reset
* Fix stack trimming
* Better errors for Queues with bad sizes
* Fix directions in dynamic indexing of complex Vecs

#### Features
* Dependency API
  * Support for specifying FIRRTL transforms via their dependencies on other transforms
  * Not yet used as specification for all transforms yet, but the API is there
* Constant Propagate muxes with literal inputs into Ands and Ors
* Provide SimplifyMems for lowering aggregate memories without splitting
* Add MultiTargetAnnotations
* Allow naming of blackbox `.f` file
* Add support for inferring the type of Reset
* Remove stack traces from user errors
* Improve support for nested Stage/Phase compilers
* Rename "Gender" to "Flow"
* Add basic support for `read-under-write` memories
* Improve API of `InstanceGraph`, used boxed strings

#### Bugfixes
* Improved stability of analog attachments
* Fix renaming of non-local annotation targets
* Fix TopWiringTransform
* Fix RemoveReset when init is invalid
* Fix annotation renaming in InlineInstances
* Make `TopWiringTransform` idempotent

---------

**Related issue**: 

**Type of change**: other enhancement

**Impact**: API addition (no impact on existing code)

**Development Phase**: implementation

**Release Notes**
* Use native chisel3 async reset support instead of blackboxed support

### CODE REVIEW

**Comment by jackkoenig** [comment]
Don't merge until further testing and I need to write up the summary of changes to chisel and firrtl included in this bump

File: src/main/scala/util/AsyncResetReg.scala [hunk 1]
```diff
@@ -54,26 +55,15 @@ class SimpleRegIO(val w: Int) extends Bundle{
...
+  val reg = withReset(reset.asAsyncReset)(RegInit(init.U(w.W)))
+  when (io.en) {
+    reg := io.d
   }
-
-  io.q := q.asUInt
-
-  override def desiredName = s"AsyncResetRegVec_w${w}_i${init}"
-
+  io.q := reg
```
Comment (aswaterman) [inline src/main/scala/util/AsyncResetReg.scala]: Omg it’s so much better

**Comment by hcook** [comment]
Can you open an issue with the list of utility modules you believe could or should be refactored or removed by taking advantage of the new features, particularly AsyncReset?

File: src/main/scala/util/AsyncResetReg.scala (hunk 1 above)
Comment (mwachs5) [reply src/main/scala/util/AsyncResetReg.scala]: not that we used it much, but what about the enable signal?

**Pull request merged**